packages/react/cypress/component/auto/form/AutoHasOneForm.cy.tsx

```diff
@@ -16,177 +12,184 @@ const originalDoodadLinkedToWidget = {
+                edges: gizmos,
+                __typename: "SectionConnection",
+              },
+              gadgetMeta: {
+                hydrations: {
+                  createdAt: "DateTime",
+                  updatedAt: "DateTime",
+                },
+                __typename: "GadgetApplicationMeta",
               },
-              __typename: "GadgetApplicationMeta",
             },
-          },
-        });
-      }
-    ).as("gizmos");
-  };
-
-  beforeEach(() => {
-    cy.viewport("macbook-13");
+          });
+        }
+      ).as("gizmos");
+    };
 
-    interceptModelUpdateActionMetadata();
-    interceptGizmosOptionsQuery();
-    interceptWidgetQuery();
-  });
+    beforeEach(() => {
+      cy.viewport("macbook-13");
 
-  it("renders form fields for related record", () => {
-    cy.mountWithWrapper(
-      <PolarisAutoForm action={api.widget.update} findBy="42">
-        <PolarisSubmitResultBanner />
-        <PolarisAutoHasOneForm
-          field="doodad"
-          primaryLabel="name"
-          secondaryLabel={(record) => `Weight:${record.weight} (${record.active})`}
-          tertiaryLabel="size"
-        >
-          <PolarisAutoInput field="name" />
-          <PolarisAutoInput field="weight" />
-          <PolarisAutoInput field="active" />
-          <PolarisAutoInput field="size" />
-        </PolarisAutoHasOneForm>
-        <PolarisAutoSubmit id="submit" />
-      </PolarisAutoForm>,
-      PolarisWrapper
-    );
+      interceptModelUpdateActionMetadata();
+      interceptGizmosOptionsQuery();
+      interceptWidgetQuery();
+    });
 
-    cy.wait("@ModelCreateActionMetadata");
-    cy.wait("@widget");
+    it("renders form fields for related record", () => {
+      cy.mountWithWrapper(
+        <AutoForm action={api.widget.update} findBy="42">
+          <SubmitResultBanner />
+          <AutoHasOneForm
+            field="doodad"
+            primaryLabel="name"
+            secondaryLabel={(record: any) => `Weight:${record.weight} (${record.active})`}
+            tertiaryLabel="size"
+          >
+            <AutoInput field="name" />
+            <AutoInput field="weight" />
+            <AutoInput field="active" />
+            <AutoInput field="size" />
+          </AutoHasOneForm>
+          <AutoSubmit id="submit" />
+        </AutoForm>,
+        wrapper
+      );
 
-    cy.contains("Doodad 1"); // primary label
-    cy.contains("Weight:333"); // secondary label
-    cy.contains("Large"); // tertiary label
+      cy.wait("@ModelCreateActionMetadata");
+      cy.wait("@widget");
 
-    cy.get(".Polaris-Button__Icon").first().click();
-    cy.contains("Edit doodad").click();
+      cy.contains("Doodad 1"); // primary label
+      cy.contains("Weight:333"); // secondary label
+      cy.contains("Large"); // tertiary label
 
-    cy.get('input[id="widget.doodad.name"]').should("exist").click().type(" - updated");
-    cy.get('input[id="widget.doodad.weight"]').should("exist").click().type("123");
```

Comment:
Polaris was able to open a modal which had "Har" and type "ry" which resulted in "Harry" while Shadcn Modal open resulted in the first input taking autofocus, so the click and type was replace the first element with the other element . e.g "Har" and type "ry" was resulting in "ry". Real Users in that case will click at the end of the input to keep typing so it won't be an issue. below is what the UI looked like when a user opens the dialog
<img width="533" alt="Screenshot 2025-02-04 at 16 49 41" src="https://github.com/user-attachments/assets/8697c798-6224-499d-9a5b-0b3aed948539" />